src/modules/TodoListArray.js

```diff
@@ -0,0 +1,53 @@
+class TodoListArray {
+  constructor() {
+    this.todos = JSON.parse(localStorage.getItem('todos')) || [];
+  }
+
+  addTodo(todo) {
+    this.todos.push(todo);
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  updateTodo(idx, newTodo) {
+    this.todos[idx - 1] = newTodo;
+    for (let i = 0; i < this.todos.length; i += 1) {
+      this.todos[i].idx = i + 1;
+    }
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  deleteTodo(index) {
+    if (this.todos.length === 1) {
+      this.todos = [];
+    } else {
+      this.todos.splice(index, 1);
+    }
+    for (let i = 0; i < this.todos.length; i += 1) {
+      this.todos[i].idx = i + 1;
+    }
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
```

Comment:
- [ ] seems your code contains repetition when it comes to updating the `todos` array, I would suggest using a helper function to do that task for you.

    eg: 
    ```javascript
    const update = (todos) => {
        todos.forEach(todo => {todo.id +=1;});
        localStorage.setItem('todos', JSON.stringify(this.todos));
    }
    ```